### PR TITLE
Features for getting submission details

### PR DESCRIPTION
**Pull request by boianradu**
SubmissionDetails struct 
GetSubmission function
Refactoring
Fixes according to the comments

### CODE REVIEW

**Comment by boianradu** [comment]
New pull request @emilva

File: v1/submissions.go [hunk 1]
```diff
@@ -144,6 +145,229 @@ type Submission struct {
 	CloseReason string
 }
 
+/*
+	INTIGRITI Report structure
```
Comment (emilva) [inline v1/submissions.go]: I think we can remove this comment, and for reference to enable Go Doc, comments should follow this standard: https://tip.golang.org/doc/comment but I do not think it is necessary here

File: v1/submissions.go [hunk 2]
```diff
@@ -144,6 +145,229 @@ type Submission struct {
...
+			Value string `json:"value"`
+		} `json:"type"`
+		Reference string `json:"reference"`
+		URL       string `json:"url"`
+	} `json:"integrationReferences"`
+}
+
+/*
+	Attachment structure used in Submission
+*/
```
Comment (emilva) [inline v1/submissions.go]: delte comment

File: v1/submissions.go [hunk 3]
```diff
@@ -174,7 +398,8 @@ func (e *Endpoint) GetSubmissions() ([]Submission, error) {
 		return findings, errors.Wrap(err, "could not authenticate to intigriti API")
 	}
 
-	req, err := http.NewRequest(http.MethodGet, e.URLApiSubmissions, nil)
+	// req, err := http.NewRequest(http.MethodGet, e.URLApiSubmissions, nil) //ORIGINAL
```
Comment (emilva) [inline v1/submissions.go]: delete comment

File: v1/submissions.go [hunk 4]
```diff
@@ -250,3 +475,57 @@ func (e *Endpoint) GetSubmissions() ([]Submission, error) {
 
 	return findings, nil
 }
+
+/*
+	GetSubmission gets report/submission by code/id
```
Comment (emilva) [inline v1/submissions.go]: remove

File: v1/submissions.go [hunk 5]
```diff
@@ -250,3 +475,57 @@ func (e *Endpoint) GetSubmissions() ([]Submission, error) {
...
+*/
+func (e *Endpoint) GetSubmission(code string) (*SubmissionDetails, error) {
+	var submi SubmissionDetails
+	var respBytes []byte
+	var err error
+	var req *http.Request
+	url := apiEndpointV1 + "/SCHIBSTED-LKJ2GIWL"
```
Comment (emilva) [inline v1/submissions.go]: This looks hard coded ;)

File: v1/submissions.go [hunk 6]
```diff
@@ -250,3 +475,57 @@ func (e *Endpoint) GetSubmissions() ([]Submission, error) {
...
+	if resp.StatusCode > 399 {
+		return nil, errors.Errorf("fetch from intigriti returned status code: %d", resp.StatusCode)
+	}
+
+	respBytes, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not read response")
+	}
+	err = json.Unmarshal(respBytes, &submi)
+	if err != nil && submi.Code == "" {
```
Comment (emilva) [inline v1/submissions.go]: what is this `submi.Code` special case? if the err is not nil, then submi will be empty?

Comment (boianradu) [reply v1/submissions.go]: I got an error for a specific report on the substructure of SubmissionDetails for the field Events in which the subfield To was different or empty than the one in the structure. The rest of the structure was unmarshalled correctly. 

I didn't dig up more to find a better structure that fits this case. From experience, it seems we'll have to use an interface for the events and declare a separate structure in which we can map the new To interface. That will change many things and the workflow

File: v1/submissions.go (hunk 5 above)
Comment (boianradu) [reply v1/submissions.go]: :(
I had to test the case for the following comment...  I'm not trying to explain the mistake, it's my bad, I just want to show the connection between them

**Review by emilva — APPROVED** [review]

**Pull request merged**